File: frontend/src/services/ReviewService.js
import reviewApi from "../api/ReviewApi";
import {parsePaginatedResponse} from "../utils/ResponseUtils";
import mediaApi from "../api/MediaApi";
import userApi from "../api/UserApi";

const ReviewService = (() => {
    const getReviewById = async (id) => {
        const response = await reviewApi.getReviewById(id);
        return response;
    };

    const getReviewsByMediaId = async (mediaId,page= 1) => {
        const res = await reviewApi.getReviewsByMediaId(mediaId);
        return parsePaginatedResponse(res);
    };

    const getReviewsByMediaIdandUserId = async (mediaId,userId) => {
        const res = await reviewApi.getReviewsByMediaIdandUserId(mediaId,userId);
        return res;
    }

    const getMovieReviewsFromUser = async (username,page= 1) => {
        const res = await reviewApi.getMovieReviewsFromUser(username,page);
        return parsePaginatedResponse(res);
    }

    const editReview = async (mediaId,rating, reviewContent) => {
        const res = await reviewApi.editReview({mediaId,rating,reviewContent});
        return res;
    }

    const createReview = async (mediaId,rating, reviewContent) => {
        const res = await reviewApi.createReview({mediaId,rating,reviewContent});
        return res;
    }

    const deleteReviewById = async (id) => {
        const response = await reviewApi.deleteReviewById(id);
        return  response;
    };

    const likeReview = async (id) => {
        const response = await reviewApi.likeReview(id);
        return  response;

    };

    return {
        getReviewById,
        getReviewsByMediaId,
        getReviewsByMediaIdandUserId,
        getMovieReviewsFromUser,
        editReview,
        createReview,
        deleteReviewById,
        likeReview,
    };
})();

export default ReviewService;

File: interfaces/src/main/java/ar/edu/itba/paw/services/ReviewService.java
package ar.edu.itba.paw.services;

import ar.edu.itba.paw.models.Review.MoovieListReview;
import ar.edu.itba.paw.models.Review.Review;
import ar.edu.itba.paw.models.Review.ReviewTypes;

import java.util.List;

public interface ReviewService {
    Review getReviewById(int reviewId);
    //Get all the reviews for a media, contains all nescesary  data to display
    List<Review> getReviewsByMediaId(int mediaId, int size, int pageNumber) ;
    int getReviewsByMediaIdCount(int mediaId);
    //Returns all the reviews a user has made, contains all nescesary data to display
    List<Review> getMovieReviewsFromUser(int userId, int size, int pageNumber);

    MoovieListReview getMoovieListReviewById(int moovieListReviewId);
    List<MoovieListReview> getMoovieListReviewsByMoovieListId(int moovieListId, int size, int pageNumber) ;
    int getMoovieListReviewByMoovieListIdCount(int moovieListId);
    List<MoovieListReview> getMoovieListReviewsFromUser(int userId, int size, int pageNumber);


    //The following work for both MoovieListsReviews and Reviews
    void createReview(int mediaId, int rating, String reviewContent, ReviewTypes type);

    void editReview(int mediaId, int rating, String reviewContent, ReviewTypes type);
    void deleteReview(int reviewId, ReviewTypes type);

    boolean likeReview(int reviewId, ReviewTypes type);
    void removeLikeReview(int reviewId, ReviewTypes type);

    Review getReviewByMediaIdAndUsername(int mediaId, int userId);
    MoovieListReview getMoovieListReviewByListIdAndUsername(int listId, int userId);

}
